### PR TITLE
fix: upload all dSYMs including those generated by other Fastlane actions

### LINKED ISSUE #51: upload_symbols_to_bugsnag only works for me when I add the version and build code
### Describe the bug
upload_symbols_to_bugsnag only works for me when I add the version and build code. I ended up having some huge versions 1.8.151445

upload_symbols_to_bugsnag is only based on getting the app version, it should be able to work using the build version

Captura de pantalla 2021-03-16 a las 10 09 13

### Steps to reproduce
if I upload the dsym like this, it generally doesn't work.
The UUID of the dsym appears to me uploaded but the dsym of the version that I just uploaded appears to me without uploading
```
download_dsyms(version: ENV["VERSION"], build_number: ENV["VERSION_CODE"], wait_for_dsym_processing: true)       # Download dSYM files from iTC
upload_symbols_to_bugsnag
```

If I do it like this it usually works
```
download_dsyms(version: ENV["VERSION"] + ENV["VERSION_CODE"], build_number: ENV["VERSION_CODE"], wait_for_dsym_processing: true)       # Download dSYM files from iTC
upload_symbols_to_bugsnag
```

### Environment
* fastlane:
    * fastlane version: 2.178.0

### LINKED ISSUE #66: uploadSymbolsToBugsnag() is not detecting dsymPath
Hi there, 

is Swift officially supported now? `uploadSymbolsToBugsnag()` is not detecting the `dsymPath` in my setup. Neither the files in the current directory nor through invocation of `downloadDsyms` before.

```
func refreshDsymsLane() {
    downloadDsyms(username: "m***@***.com", appIdentifier: "com.vjeton.diamond")
    uploadSymbolsToBugsnag(apiKey: "71b**************867")
    cleanBuildArtifacts()
  }
```
 
Result:

```
INFO [2021-08-26 22:24:20.10]: ---------------------------------------
INFO [2021-08-26 22:24:20.10]: --- Step: upload_symbols_to_bugsnag ---
INFO [2021-08-26 22:24:20.10]: ---------------------------------------
DEBUG [2021-08-26 22:24:20.10]: Uploading dSYMs to Bugsnag with the following parameters:
DEBUG [2021-08-26 22:24:20.10]:              api_key: 71b**************867
DEBUG [2021-08-26 22:24:20.10]:            dsym_path: []
DEBUG [2021-08-26 22:24:20.10]:         project_root: /Users/mario/Repository/VJeton
DEBUG [2021-08-26 22:24:20.10]:           upload_url:
DEBUG [2021-08-26 22:24:20.10]:     symbol_maps_path:
DEBUG [2021-08-26 22:24:20.10]:   ignore_missing_dwarf: false
DEBUG [2021-08-26 22:24:20.10]:    ignore_empty_dsym: false
DEBUG [2021-08-26 22:24:20.10]:          config_file:
DEBUG [2021-08-26 22:24:20.10]: return_value_type is nil value: []
DEBUG [2021-08-26 22:24:20.10]: sending {"payload":{"status":"ready_for_next","return_object":[],"closure_argument_value":""}}
DEBUG [2021-08-26 22:24:20.10]: received done signal shutting down

+------+---------------------------+-------------+
|                fastlane summary                |
+------+---------------------------+-------------+
| Step | Action                    | Time (in s) |
+------+---------------------------+-------------+
| 1    | download_dsyms            | 20          |
| 2    | upload_symbols_to_bugsnag | 0           |
+------+---------------------------+-------------+
```

When I try to set it manually, it works. Example:

```
uploadSymbolsToBugsnag(
      apiKey: "71b**************867",
      dsymPath: ["com.vjeton.diamond-1.0-25.dSYM.zip"]
    )
```

What I find especially strange is that when I call `fastlane action upload_symbols_to_bugsnag` the files are actually detected just fine:

```
+----------------------+---------------------------+---------------------------+---------------------------+
|                                    upload_symbols_to_bugsnag Options                                     |
+----------------------+---------------------------+---------------------------+---------------------------+
| Key                  | Description               | Env Var(s)                | Default                   |
+----------------------+---------------------------+---------------------------+---------------------------+
| api_key              | Bugsnag API Key           | BUGSNAG_API_KEY           |                           |
| dsym_path            | Path to the DSYM          | BUGSNAG_DSYM_PATH         | ["./com.vjeton.diamond-1  |
|                      | directory, file, or zip   |                           | .0-17.dSYM.zip",          |
|                      | to upload                 |                           | "./com.vjeton.diamond-1.  |
|                      |                           |                           | 0-18.dSYM.zip",           |
|                      |                           |                           | "./com.vjeton.diamond-1.  |
|                      |                           |                           | 0-19.dSYM.zip",           |
|                      |                           |                           | "./com.vjeton.diamond-1.  |
|                      |                           |                           | 0-20.dSYM.zip",           |
|                      |                           |                           | "./com.vjeton.diamond-1.  |
|                      |                           |                           | 0-21.dSYM.zip",           |
|                      |                           |                           | "./com.vjeton.diamond-1.  |
|                      |                           |                           | 0-22.dSYM.zip",           |
|                      |                           |                           | "./com.vjeton.diamond-1.  |
|                      |                           |                           | 0-23.dSYM.zip",           |
|                      |                           |                           | "./com.vjeton.diamond-1.  |
|                      |                           |                           | 0-25.dSYM.zip",           |
|                      |                           |                           | "./com.vjeton.diamond-1.  |
|                      |                           |                           | 0.1-26.dSYM.zip",         |
|                      |                           |                           | "./com.vjeton.diamond-1.  |
|                      |                           |                           | 0.1-27.dSYM.zip",         |
|                      |                           |                           | "./com.vjeton.diamond-1.  |
|                      |                           |                           | 0.1-28.dSYM.zip"]
```

It'd be really convenient for me if the files could be detected automatically. :)

Kind regards
Mario

### PR DESCRIPTION
**Pull request by xander-jones**
# Goal

Upload all the files! Upload dSYMs scraped from local directory, dSYMs pointed to by Fastlane actions `gym`, and `download_dsyms`. Previously, we had an order of precedence, whereby if one of these actions ran, it would overrule the other.

Fixes #66, and should also fix #51 (due to SharedValue overriding the scraped path).

## Design

If the constants as set by `gym` or `download_dsyms` exist, and the key exists in the lane_context set by Fastlane, append these arrays to one big dsym array to parse. Rejects paths which begin `./vendor`, as this will pick up the `rspec` dSYMs which should not be uploaded.

Add these values as output diagnostics when run verbosely to help with support and debugging.

## Changeset

* `upload_symbols_to_bugsnag` action updated
* Automated tests updated to suit
* `dsym_magic` removed as this functionality is now in the `Fastfile`

## Testing

Automated tests added to show the SharedValues when set are picked up and uploaded as expected.

### CODE REVIEW

File: features/fixtures/fl-project/fastlane/Fastfile [hunk 1]
```diff
@@ -23,10 +23,24 @@ platform :ios do
     upload_symbols_to_bugsnag(upload_url: endpoint, api_key: api_key, ignore_empty_dsym: ignore_empty_dsym, ignore_missing_dwarf: ignore_missing_dwarf)
   end
 
-  lane :upload_symbols_with_custom_action do
-    dsym_magic # Sets a custom value to DSYM_OUTPUT_PATH
-    upload_symbols_to_bugsnag(upload_url: endpoint, api_key: api_key)
+  lane :upload_symbols_after_gym do
+    # Sets a custom value to DSYM_OUTPUT_PATH
+    Actions.lane_context[SharedValues::DSYM_OUTPUT_PATH] = ["../app-center"]
```
Comment (kattrali) [inline features/fixtures/fl-project/fastlane/Fastfile:28]: Is this value always set to an array?

File: tools/fastlane-plugin/lib/fastlane/plugin/bugsnag/actions/upload_symbols_to_bugsnag.rb [hunk 2]
```diff
@@ -41,7 +44,7 @@ def self.description
       end
 
       def self.authors
-        ["kattrali"]
+        ["kattrali", "xander-jones"]
```
Comment (kattrali) [inline tools/fastlane-plugin/lib/fastlane/plugin/bugsnag/actions/upload_symbols_to_bugsnag.rb:47]: 😎

File: features/fixtures/fl-project/fastlane/Fastfile (hunk 1 above)
Comment (xander-jones) [reply features/fixtures/fl-project/fastlane/Fastfile:28]: No, in fact it's never set to an array, it should be just a plain path according to the [docs](https://docs.fastlane.tools/actions/gym/#lane-variables), that needs fixing :)

On the flipside the SharedVariable of `download_dsyms` is always an array: https://docs.fastlane.tools/actions/download_dsyms/#lane-variables

Comment (xander-jones) [reply features/fixtures/fl-project/fastlane/Fastfile]: Fixed in [`ce9286f` (#67)](https://github.com/bugsnag/bugsnag-dsym-upload/pull/67/commits/ce9286f0dfd76d37f86e9efcf16de3412c8fa221)

Added new function to coerce into an Array if required. The `Dir` function should always return an array though, so not put this through the new function.

**Review by kattrali — APPROVED** [review]

**Pull request merged**